packages/ui-components/src/components/absolute-time-picker-dropdown-input/absolute-time-picker-dropdown-input.tsx

```diff
@@ -138,18 +157,24 @@ export class KvAbsoluteTimePickerDropdownInput implements IAbsoluteTimePickerDro
 		if (this.mode === EAbsoluteTimePickerMode.Range) this.hoveredDate = event.detail;
 	};
 
-	private handleOutsideClick = () => {
+	private handleCloseDropdown = () => {
 		this.isDropdownOpen = false;
 		this.focusedInput = null;
 		if (this.mode === EAbsoluteTimePickerMode.Single) {
-			!isEmpty(this.singleTimeInputValue) && this.selectedTimeChange.emit([dayjs(this.singleTimeInputValue, DATE_FORMAT).tz(this.timezone.name, true).valueOf()]);
+			if (isEmpty(this.singleTimeInputValue)) {
+				this.syncSelectedTimeState(this.selectedTime);
+			} else {
+				this.selectedTimeChange.emit([dayjs(this.singleTimeInputValue, DATE_FORMAT).tz(this.timezone.name, true).valueOf()]);
+			}
 		} else {
-			!isEmpty(this.rangeInputValues?.from) &&
-				!isEmpty(this.rangeInputValues?.to) &&
+			if (isEmpty(this.rangeInputValues?.from) || isEmpty(this.rangeInputValues?.to)) {
+				this.syncSelectedTimeState(this.selectedTime);
+			} else {
 				this.selectedTimeChange.emit([
 					dayjs(this.rangeInputValues.from, DATE_FORMAT).tz(this.timezone.name, true).valueOf(),
```

Comment:
Hmmm, this implementation can introduce an error in the process if the `this.rangeInputValues.from` or `this.rangeInputValues.to` are empty? Or not because the dayjs on undefined will generate the current day?

Reply:
It cannot. I do that validation to check if it is empty. If not I also validate the date with the dayjs isValid method.

Reply:
Furthermore, I only emit when the dropdown is closed. If one of the values is empty when the dropdown is closed, the expected behaviour is to set the previously set values because the user has not introduced a new valid date.